Modules/Test/test/ilTestEvaluationUserDataTest.php

```diff
@@ -346,13 +346,32 @@ public function testGetQuestionCount(): void
         $this->assertEquals(5, $this->testObj->getQuestionCount());
     }
 
-    public function testReachedPoints(): void
+    /**
+     * @dataProvider reachedPointsDataProvider
+     */
+    public function testReachedPoints($input, float $expected): void
+    {
+        $pass = new ilTestEvaluationPassData();
+        $pass->setReachedPoints($input);
+        $this->testObj->addPass(0, $pass);
+
+        $this->assertEquals($expected, $this->testObj->getReachedPoints());
+    }
+
+    public function reachedPointsDataProvider(): array
+    {
+        return [
+            "float" => [25.0, 25.0],
```

Comment:
To prevent future issues and address the bug description effectively, I recommend adding an additional test case to cover floating point cutoff resulting from integer cast.

```suggestion
            "float X.0" => [25.0, 25.0],
            "float X.Y" => [25.3, 25.3],
```